onnx/backend/test/runner/__init__.py

```diff
@@ -288,28 +288,33 @@ def run(test_self, device):  # type: (Any, Text) -> None
                 self.assert_similar_outputs(ref_outputs, outputs,
                                             rtol=model_test.rtol,
                                             atol=model_test.atol)
-
             for test_data_dir in glob.glob(
                     os.path.join(model_dir, "test_data_set*")):
                 inputs = []
                 inputs_num = len(glob.glob(os.path.join(test_data_dir, 'input_*.pb')))
                 for i in range(inputs_num):
                     input_file = os.path.join(test_data_dir, 'input_{}.pb'.format(i))
-                    tensor = onnx.TensorProto()
-                    with open(input_file, 'rb') as f:
-                        tensor.ParseFromString(f.read())
-                    inputs.append(numpy_helper.to_array(tensor))
+                    self._load_proto(input_file, inputs, model.graph.input[i].type)
```

Comment:
Just thinking out aloud - does this logic of iterating through input/output proto files handle optional inputs/outputs ? Let us say we have input_0.pb and input_2.pb (with input 1 optional), does it break then ?

Reply:
https://github.com/onnx/onnx/pull/3136#discussion_r534397071
Just like what @gramalingam said above, the input/output protobuf files are generated by the model.graph in the same order so I think it should be fine? If the input or output is optional, then it should not show in the model.graph either.